--- /usr/share/webapps/rutorrent/php/Snoopy.class.inc	2017-02-24 13:09:52.700398089 +0000
+++ /usr/share/webapps/rutorrent/php/Snoopy.class.inc	2017-02-24 13:10:25.736389184 +0000
@@ -317,7 +317,7 @@
 		if(!empty($this->agent))
 			$headers .= "User-Agent: ".$this->agent."\r\n";
 		if(!empty($this->host) && !isset($this->rawheaders['Host']))
-			$headers .= "Host: ".$this->host.($this->port == 80 ? "" : ":".$this->port)."\r\n";
+			$headers .= "Host: ".$this->host.":".$this->port."\r\n";
 		if(!empty($this->accept))
 			$headers .= "Accept: ".$this->accept."\r\n";
 		if($this->use_gzip && function_exists("gzinflate")) 
@@ -454,7 +454,7 @@
 		if(!empty($this->agent))
 			$headers[] = "User-Agent: ".$this->agent;
 		if(!empty($this->host) && !isset($this->rawheaders['Host']))
-			$headers[] = "Host: ".$this->host.($this->port == 443 ? "" : ":".$this->port);
+			$headers[] = "Host: ".$this->host;
 		if(!empty($this->accept))
 			$headers[] = "Accept: ".$this->accept;
 		if(!empty($this->referer))
